relationships.md

```diff
@@ -12,14 +12,14 @@ A sample can be found at https://dspace7-entities.atmire.com/rest/#https://dspac
 ## Single Relationship
 **/api/core/relationships/<:id>**
 
-A sample can be found at https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/relationships/530
-The leftId and rightId parameter in JSON may still be present, but they will be removed soon (requires other PRs to be merged first)
+A sample can be found at https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/relationships/1117
 
 ```json
 {
   "id": 530,
   "relationshipTypeId": 0,
   "leftPlace": 1,
+  "leftwardLabel": "Name variant 1",
```

Comment:
Thank you @benbosman! 
As I understood, there will be both a leftwardLabel and rightwardLabel on relationship_type (renaming the existing ones) and here, on relationship. 
Will this leftwardLabel or rightwardLabel override what is defined by the relationship_type? I'm confused regarding the chosen names. What will held what? And I think others will also get confused. Perhaps it's best to get a different property name for the relationship object like a relation attribute (something like leftRelationAttribute or rightRelationAttribute). 
The example shows `"relationshipTypeId": 0` but there isn't any particular reason to be 0, right? The relation type it's always mandatory, even with "labels".

Reply:
@paulo-graca 
* The leftwardLabel on the relationship type is e.g. isAuthorOfPublication, it is the label used to describe the type of relationship
* The leftwardLabel on the relationship is e.g. "Donald, Mike", it is the label used to describe the left item (or the name variant)

I didn't realize there was still a `relationshipTypeId` in the relationship implementation. This must be a left-over from when it was not a HAL link yet. This will need to be removed from the implementation (it's not used or populated anyway)

Reply:
Ok, thank you @benbosman for enlighten me! But it's a "label" (at the relationship level) or it's a relationship "attribute" or "property"? This might influence the naming.

Reply:
@paulo-graca it's a property of the relationship.
The reason why I used the term label is because it labels the left or right item from the point of view of the other item.
Would it be clearer to use `leftwardNameVariant` and `rightwardNameVariant`. Or would this make too many assumptions and make it less usable when used for non-person objects.

Reply:
Do you agree with this suggestion?
Replacing of `relationship.leftwardLabel`  with `relationship.leftwardProperty` and `relationship.rightwardLabel`  with `relationship.rightwardProperty` and leave `relationship_type.leftwardLabel` and `relationship_type.rightwardLabel` as was suggested. My point of view is that those relationship values will have a different purpose than what is used on the relationship_type.

Reply:
The actual name doesn't matter too much for me, I'm probably a bit too closely involved 
In DSpace 7 it would be used for name variants only (but it supports name variants for e.g. an Org Unit as well)
In DSpace 8 it may be expanded to support metadata completely

@tdonohue what name would you consider most suitable since you have already reviewed this PR?

Reply:
As I noted in Slack, I understand @paulo-graca's point here.  The word "label" is seemingly being used for two different purposes. In the case of "relationship_type" the label describes the *type* of the relationship (e.g. "isAuthorOfPublication").  In the case of "relationship" the label is almost like metadata (e.g. "Smith, Jane"), or could be considered the "value" of the type "isAuthorOfPublication" (as essentially we are saying here "isAuthorOfPublication = Smith, Jane").

I admit, I don't feel strongly either way (as this could be described in documentation to clarify).  But, if we were to go this route, I wonder if we should drop the word "label" altogether and replace it with better terms.

For example:
* relationship_type.leftwardType = "isAuthorOfPublication"   (This is the Type of the Relationship)
* relationship.leftwardValue = "Smith, Jane" (This is the value of the Type of the Relationship)

leftwardValue could also be called leftwardMetadata (but metadata might be too generic of a term too).

Reply:
That's more less confusing to me!